lib/ecto_diff.ex

```diff
@@ -367,7 +367,47 @@ defmodule EctoDiff do
     end
   end
 
-  defp diff_association(previous, current, %{cardinality: :many, field: field, related: struct}, acc, opts) do
+  defp diff_association(
+         previous,
+         current,
+         %{cardinality: :many, through: [through, assoc_field], owner: owner},
+         acc,
+         opts
+       ) do
+    assoc_diff =
+      acc
+      |> Keyword.get_values(through)
+      |> List.flatten()
+      |> Enum.map(& &1[:changes][assoc_field])
```

Comment:
I wonder if it's possible to replace this `List.flatten + Enum.map` with a `Enum.flat_map`

Reply:
`Enum.flat_map/2` flattens the result of the operation, not the first argument given to it.